### PR TITLE
wire cleardata method from the database to the StorageServiceManager

### PR DESCRIPTION
**Pull request by galganif**

### CODE REVIEW

File: java/arcs/core/storage/DriverFactory.kt [hunk 1]
```diff
@@ -48,6 +51,20 @@ object DriverFactory {
             ?.getDriver(storageKey, dataClass, type)
     }
 
+    suspend fun removeAllEntities(): Job = coroutineScope {
+        launch {
+            providers.value.forEach { it.removeAllEntities() }
+        }
+    }
+    suspend fun removeEntitiesCreatedBetween(startTimeMillis: Long, endTimeMillis: Long): Job =
```
Comment (jasonwyatt) [inline java/arcs/core/storage/DriverFactory.kt:59]:
```suggestion
    }
    
    suspend fun removeEntitiesCreatedBetween(startTimeMillis: Long, endTimeMillis: Long): Job =
```

File: java/arcs/core/storage/DriverFactory.kt [hunk 2]
```diff
@@ -69,4 +86,9 @@ interface DriverProvider {
         dataClass: KClass<Data>,
         type: Type
     ): Driver<Data>
+
+    // TODO: once all DriverProviders implement this, we can remove these defaults.
+    suspend fun removeAllEntities() {}
```
Comment (jasonwyatt) [inline java/arcs/core/storage/DriverFactory.kt:91]:
```suggestion
    suspend fun removeAllEntities() = Unit
```

File: java/arcs/core/storage/DriverFactory.kt [hunk 3]
```diff
@@ -69,4 +86,9 @@ interface DriverProvider {
...
+
+    suspend fun removeEntitiesCreatedBetween(startTimeMillis: Long, endTimeMillis: Long) {}
```
Comment (jasonwyatt) [inline java/arcs/core/storage/DriverFactory.kt:93]:
```suggestion
    suspend fun removeEntitiesCreatedBetween(startTimeMillis: Long, endTimeMillis: Long) =
        Unit
```

File: java/arcs/core/storage/driver/Database.kt [hunk 4]
```diff
@@ -92,6 +92,12 @@ object DatabaseDriverProvider : DriverProvider {
         ).register()
     }
 
+    override suspend fun removeAllEntities() { manager.removeAllEntities() }
```
Comment (jasonwyatt) [inline java/arcs/core/storage/driver/Database.kt:95]: Either use function-expression notation, or new lines between braces and the function body.
```suggestion
    override suspend fun removeAllEntities() { 
        manager.removeAllEntities() 
    }
```

File: javatests/arcs/android/storage/service/StorageServiceManagerTest.kt [hunk 5]
```diff
@@ -11,29 +11,112 @@
...
+        manager.clearDataBetween(2,2, deferredResult)
+
+        assertThat(deferredResult.await()).isTrue()
+        assertThat(handle.fetchAll()).containsExactly(entity1, entity3)
     }
+
+    private suspend fun createSingletonHandle() =
+        EntityHandleManager(
+            time = FakeTime(),
+            scheduler = JvmSchedulerProvider(EmptyCoroutineContext).invoke("test")
```
Comment (jasonwyatt) [inline javatests/arcs/android/storage/service/StorageServiceManagerTest.kt:92]: Maybe share the scheduler provider? (or the whole handle manager?)

**Comment by galganif** [comment]
Thanks, all done.

**Pull request merged**